problem building with perlmutter system hdf5

Build issue on Perlmutter using spack:

```
-- Looking for HDF5 at: /opt/cray/pe/hdf5/1.12.1.1/gnu/9.1
-- Found HDF5: Included by compiler wrappers (found version "1.12.1")
-- HDF5_DIR_REAL=/opt/cray/pe/hdf5/1.12.1.1/gnu/9.1
-- Checking that found HDF5_INCLUDE_DIRS are in HDF5_DIR
CMake Error at cmake/thirdparty/SetupHDF5.cmake:77 (message):
  FindHDF5 did not provide HDF5_INCLUDE_DIRS or HDF5_INCLUDE_DIR.
Call Stack (most recent call first):
  cmake/Setup3rdParty.cmake:101 (include)
  CMakeLists.txt:83 (include)
```

Newer CMake or HDF5 may not be providing the same cmake vars that we rely on for sanity checks.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,7 +14,7 @@ and this project aspires to adhere to [Semantic Versioning](https://semver.org/s
 ### Changed
 
 #### Relay
-- Changed HDF5 CMake sanity checks to issue `WARNING` instead of `FATAL_ERROR`, since Cray system HDF5 installs do not present this info we use for sanity checks. 
+- Changed HDF5 CMake sanity checks to issue `WARNING` instead of `FATAL_ERROR`, since Cray system HDF5 installs do not always present the info we use for sanity checks.
 
 ### Fixed
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,11 @@ and this project aspires to adhere to [Semantic Versioning](https://semver.org/s
 #### Blueprint
 - Added `blueprint::mesh::paint_adjset`, which paints fields that encode adjacency set counts and ordering details. 
 
+### Changed
+
+#### Relay
+- Changed HDF5 CMake sanity checks to issue `WARNING` instead of `FATAL_ERROR`, since Cray system HDF5 installs do not present this info we use for sanity checks. 
+
 ### Fixed
 
 #### Blueprint

--- a/src/cmake/thirdparty/SetupHDF5.cmake
+++ b/src/cmake/thirdparty/SetupHDF5.cmake
@@ -74,12 +74,12 @@ if(NOT HDF5_INCLUDE_DIRS)
     if(HDF5_INCLUDE_DIR)
         set(HDF5_INCLUDE_DIRS ${HDF5_INCLUDE_DIR})
     else()
-        message(FATAL_ERROR "FindHDF5 did not provide HDF5_INCLUDE_DIRS or HDF5_INCLUDE_DIR.")
+        message(WARNING "FindHDF5 did not provide HDF5_INCLUDE_DIRS or HDF5_INCLUDE_DIR.")
     endif()
 endif()
 
 if(NOT HDF5_LIBRARIES)
-    message(FATAL_ERROR "FindHDF5 did not provide HDF5_LIBRARIES.")
+    message(WARNING "FindHDF5 did not provide HDF5_LIBRARIES.")
 endif()
 
 message(STATUS "HDF5_INCLUDE_DIRS=${HDF5_INCLUDE_DIRS}")
@@ -104,7 +104,7 @@ foreach(IDIR ${HDF5_INCLUDE_DIRS})
 endforeach()
 
 if(NOT check_hdf5_inc_dir_ok)
-    message(FATAL_ERROR " ${HDF5_INCLUDE_DIRS} does not include HDF5_DIR")
+    message(WARNING " ${HDF5_INCLUDE_DIRS} does not include HDF5_DIR")
 endif()
 
 #
@@ -315,6 +315,16 @@ elseif(TARGET hdf5::hdf5-static)
     message(STATUS "HDF5 using hdf5::hdf5-static target")
     blt_register_library(NAME hdf5
                          LIBRARIES hdf5::hdf5-static)
+elseif(TARGET hdf5-shared AND BUILD_SHARED_LIBS)
+    # reg shared ver of imported lib target
+    message(STATUS "HDF5 using hdf5-shared target")
+    blt_register_library(NAME hdf5
+                         LIBRARIES hdf5-shared)
+elseif(TARGET hdf5-static)
+    # reg static ver of imported lib target
+     message(STATUS "HDF5 using hdf5-static target")
+     blt_register_library(NAME hdf5
+                          LIBRARIES hdf5-static)
 elseif(TARGET hdf5)
     # legacy hdf5 CMake build system support creates an hdf5 target we use directly
     message(STATUS "HDF5 using hdf5 target")

--- a/src/cmake/thirdparty/SetupHDF5.cmake
+++ b/src/cmake/thirdparty/SetupHDF5.cmake
@@ -104,7 +104,7 @@ foreach(IDIR ${HDF5_INCLUDE_DIRS})
 endforeach()
 
 if(NOT check_hdf5_inc_dir_ok)
-    message(WARNING " ${HDF5_INCLUDE_DIRS} does not include HDF5_DIR")
+    message(WARNING "HDF5_INCLUDE_DIRS (${HDF5_INCLUDE_DIRS}) does not include HDF5_DIR")
 endif()
 
 #

--- a/src/config/conduit_setup_deps.cmake
+++ b/src/config/conduit_setup_deps.cmake
@@ -117,7 +117,9 @@ if(CONDUIT_HDF5_DIR)
         if(HDF5_INCLUDE_DIR)
             set(HDF5_INCLUDE_DIRS ${HDF5_INCLUDE_DIR})
         else()
-            message(WARNING "FindHDF5 did not provide HDF5_INCLUDE_DIRS or HDF5_INCLUDE_DIR.")
+            if(NOT Conduit_FIND_QUIETLY)
+                message(WARNING "FindHDF5 did not provide HDF5_INCLUDE_DIRS or HDF5_INCLUDE_DIR.")
+            endif()
         endif()
     endif()
 
@@ -150,7 +152,9 @@ if(CONDUIT_HDF5_DIR)
     endforeach()
 
     if(NOT check_hdf5_inc_dir_ok)
-        message(WARNING " ${HDF5_INCLUDE_DIRS} does not include HDF5_DIR")
+        if(NOT Conduit_FIND_QUIETLY)
+            message(WARNING "HDF5_INCLUDE_DIRS (${HDF5_INCLUDE_DIRS}) does not include HDF5_DIR")
+        endif()
     endif()
 
     #

--- a/src/config/conduit_setup_deps.cmake
+++ b/src/config/conduit_setup_deps.cmake
@@ -117,7 +117,7 @@ if(CONDUIT_HDF5_DIR)
         if(HDF5_INCLUDE_DIR)
             set(HDF5_INCLUDE_DIRS ${HDF5_INCLUDE_DIR})
         else()
-            message(FATAL_ERROR "FindHDF5 did not provide HDF5_INCLUDE_DIRS or HDF5_INCLUDE_DIR.")
+            message(WARNING "FindHDF5 did not provide HDF5_INCLUDE_DIRS or HDF5_INCLUDE_DIR.")
         endif()
     endif()
 
@@ -150,7 +150,7 @@ if(CONDUIT_HDF5_DIR)
     endforeach()
 
     if(NOT check_hdf5_inc_dir_ok)
-        message(FATAL_ERROR " ${HDF5_INCLUDE_DIRS} does not include HDF5_DIR")
+        message(WARNING " ${HDF5_INCLUDE_DIRS} does not include HDF5_DIR")
     endif()
 
     #

